core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -247,6 +255,25 @@ public Expression rewriteExtract(Extract node, Void context, ExpressionTreeRewri
             throw new UnsupportedOperationException("not yet implemented: " + node.getField());
         }
 
+        @Override
+        public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            String functionName = extractFunctionName(node.getName());
+            if (functionName.equals("date")) {
+                verify(node.getArguments().size() == 1);
+                Expression argument = node.getArguments().get(0);
+                Type argumentType = expressionTypes.get(NodeRef.of(argument));
```

Comment:
can `argumentType` be `null`?

(do we usually check for that?)

Reply:
No we dont. Also the map used for `expressionTypes` is `IMmutableMap` right now, and that one does not allow nulls.

Reply:
IMmutableMap doesn't allow nulls, but will still return null if no mapping found

Reply:
True. Still we do not check it in other places. Do you feel it would be better to do a check and gracefully bail out of function (default rewrite in this case)?

Reply:
A null would be a bug in the code. Every expression should have a type associated with it.

Reply:
> A null would be a bug in the code. 

of course. And it's desireable when bugs are noticed before they lead to silent correction issues (https://github.com/trinodb/trino/pull/7980)